firmware/main.cpp

```diff
@@ -0,0 +1,377 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2022 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "mbed.h"
+
+#include "main.h"
+#include "leds.h"
+#include "can-ids/CAN_IDS.h"
+
+
+#include "SWO/SWO.h"
+#include "WS2812/WS2812.h"
+#include "DebounceIn/DebounceIn.h"
+
+
+// Used to print debug info over SWO instead of UART
+SWO_Channel swo("channel");
+
+// CAN bus for reading RPM and sending out shift/DRS messages
+CAN can(PA_11, PA_12, CAN_BAUD);
+
+// EventQueue is used to run events from interrupts in a non-interrupt context
+EventQueue queue(32 * EVENTS_EVENT_SIZE);
+Thread t;
+
+// Periodically calls an ISR to update LEDs
+Ticker led_loop;
+
+// Periodically updates data to keep the ECU happy
+Ticker can_loop;
+
+// LED object, gets updated in a loop
+LEDs leds;
+
+// Steering wheel state
+SteeringWheelState state;
+
+// Buttons
+DebounceIn upshift_btn(UPSHIFT_PIN, PullUp);
```

Comment:
Is there a way to either update hw debounce interval or lock it so it doesn't send multiple times within say, 200 ms, or whatever interval? Open to doing this on the bcm, but it seems like it may be more straightforward to set on the sw board